internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixExternalViewRepository.java

```diff
@@ -32,9 +34,11 @@
  * Extend HelixBaseRoutingRepository to leverage external view data.
  */
 @BatchMode
-public class HelixExternalViewRepository extends HelixBaseRoutingRepository {
+public class HelixExternalViewRepository extends HelixBaseRoutingRepository implements IdealStateChangeListener {
   private static final Logger LOGGER = LogManager.getLogger(HelixExternalViewRepository.class);
 
+  protected volatile Map<String, Integer> resourceToIdealPartitionCountMap;
```

Comment:
I know it is the existing issue, but if possible, I would prefer to declare it as `AtomicReference`, as volatile only solves the visibility issue, thus making the change to `resourceToIdealPartitionCountMap` visible to all threads, however it doesn't guarantee an atomic update.

Reply:
good idea, let me update it.

Reply:
Are you saying we should update this variable with CAS? Is there a need for that?

If we don't do CAS, and only use the `get` and `set` functions on `AtomicReference`, then it is literally just setting and getting a volatile field inside the `AR`.

Reply:
Right, I discussed offline with Lei, there are no differences to use AR vs `volatile` in this case.